In [1]:
%matplotlib inline
import pandas as pd

In [2]:
from IPython.core.display import HTML
css = open('style-table.css').read() + open('style-notebook.css').read()
HTML('<style>{}</style>'.format(css))

In [8]:
titles = pd.read_csv('data/titles.csv')
titles.head()

title  year
0    Tasveer Mere Sanam  1996
1              Only You  1994
2  El pueblo del terror  1970
3               Machine  2007
4                  MARy  2008

### How many movies are listed in the titles dataframe?

In [3]:
import pandas as pd
titles = pd.read_csv('E:/LetsUpgrade/Day_16&17/titles.csv')
titles['title'].count()

244914

**There are 244914 titles in the dataframe**


### What are the earliest two films listed in the titles dataframe?

In [3]:
titles.sort_values('year').head(2)

title  year
193222          Miss Jerry  1894
51388   The Startled Lover  1898

**Miss Jerry and The Startled Lover are the 2 films listed earliest.**

### How many movies have the title "Hamlet"?

In [4]:
len(titles[titles.title=='Hamlet'])

20

**20 movies have Hamlet**

### How many movies are titled "North by Northwest"?

In [5]:

len(titles[titles.title=='North by NorthWest'])

0

**There are no movies with such title**

### When was the first movie titled "Hamlet" made?

In [6]:
titles[titles.title=='Hamlet'].sort_values('year').head(1)

title  year
52521  Hamlet  1910

**First movie with title Hamlet was made in 1910**

### List all of the "Treasure Island" movies from earliest to most recent.

In [7]:
titles[titles.title=='Treasure Island'].sort_values('year')

title  year
55686   Treasure Island  1918
196514  Treasure Island  1920
141822  Treasure Island  1934
77354   Treasure Island  1950
65051   Treasure Island  1972
37238   Treasure Island  1973
173051  Treasure Island  1985
27076   Treasure Island  1999

### How many movies were made in the year 1950?

In [8]:
titles[titles.year==1950].count()

title    1113
year     1113
dtype: int64

**1113 movies were made in 1950**

### How many movies were made in the year 1960?

In [9]:
titles[titles.year==1960].count()

title    1549
year     1549
dtype: int64

**1549 movies were made in 1960**

### How many movies were made from 1950 through 1959?

In [9]:
n=0
for i in titles['year']:
    if(i in range(1950,1959)):
        n=n+1
    else:
        continue
        
print("Number of Movies from 1950 to 1959 are : ",n)

Number of Movies from 1950 to 1959 are :  11419


### In what years has a movie titled "Batman" been released?

In [10]:
titles[titles.title=='Batman'].year

39349     1943
240896    1989
Name: year, dtype: int64

**Batman was released in 1943 and 1989**

### How many roles were there in the movie "Inception"?

In [13]:
import pandas as pd
cast=pd.read_csv('E:/LetsUpgrade/Day_16&17/cast.csv')
cast[cast.title=="Inception"].character.count()
          

28

**There are 28 roles in Inception**

### How many roles in the movie "Inception" are NOT ranked by an "n" value?

In [17]:
import pandas as pd
cast=pd.read_csv('E:/LetsUpgrade/Day_16&17/cast.csv')

inception = cast[cast.title == 'Inception']
len(inception[inception['n'].isnull()])

9

**9 roles are not ranked in Inception**

### But how many roles in the movie "Inception" did receive an "n" value?

In [14]:
x=cast[cast.title=="Inception"].character.count()
inception = cast[cast.title == 'Inception']
y=len(inception[inception['n'].isna()])
print(x-y)

19


**19 roles are ranked**

### Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

In [47]:
c=cast[cast.title=="North by Northwest"]
x=c['n'].count()
c.sort_values('n').head(x)

title  year               name   type  \
871426  North by Northwest  1959         Cary Grant  actor   
356707  North by Northwest  1959     Leo G. Carroll  actor   
665768  North by Northwest  1959  Robert Ellenstein  actor   
464736  North by Northwest  1959    Philip Coolidge  actor   
206471  North by Northwest  1959       Edward Binns  actor   

                 character     n  
871426  Roger O. Thornhill   1.0  
356707       The Professor   5.0  
665768               Licht  11.0  
464736           Dr. Cross  13.0  
206471      Captain Junket  15.0

### Display the entire cast, in "n"-order, of the 1972 film "Sleuth".

In [52]:
c=cast[cast.title=="Sleuth"]
c[c.year==1972].sort_values('n')

title  year            name   type          character    n
326773  Sleuth  1972   Michael Caine  actor        Milo Tindle  2.0
373876  Sleuth  1972  Alec Cawthorne  actor  Inspector Doppler  3.0

### Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

In [51]:
c=cast[cast.title=="Sleuth"]
c[c.year==2007].sort_values('n')

title  year                 name   type          character    n
326774  Sleuth  2007        Michael Caine  actor             Andrew  1.0
260242  Sleuth  2007      Kenneth Branagh  actor  Other Man on T.V.  NaN
373877  Sleuth  2007  Alec (II) Cawthorne  actor  Inspector Doppler  NaN

### How many roles were credited in the silent 1921 version of Hamlet?

In [91]:
c=cast[cast.title=="Hamlet"]
len(c[c.year==1921])


3

**3 roles were credited**

### How many roles were credited in Branagh’s 1996 Hamlet?

In [77]:
c=cast[cast.title=="Hamlet"]
x=c[c.year==1996]
y=x[x.name=='Kenneth Branagh']
y.n

260225    5.0
Name: n, dtype: float64

**5 roles were credited**

### How many "Hamlet" roles have been listed in all film credits through history?

In [90]:
len(cast[cast.title=="Hamlet"])


117

**117 movies have been listed** 

### How many people have played an "Ophelia"?

In [89]:
len(cast[cast.character=="Ophelia"])


4

**4 people have played Ophelia**

### How many people have played a role called "The Dude"?

In [88]:
len(cast[cast.character=="The Dude"])


7

**7 people have played the role "The Dude"**

### How many people have played a role called "The Stranger"?

In [96]:
len(cast[cast.character=="The Stranger"])

84

**84 people have played role "The Stranger"**

### How many roles has Sidney Poitier played throughout his career?

In [94]:
len(cast[cast.character=='Sidney Poitier'])

1

**Sidney has played 1 role**

### How many roles has Judi Dench played?

In [102]:
len(cast[cast.name=='Judi Dench'])

0

**No roles are played**

### List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

In [119]:
c=cast[cast.name=='Cary Grant']
x=c[(c['year'] >= 1940) & (c['year'] <= 1949)]
x[x['n']==2].sort_values('year')


title  year        name   type    character    n
871423  My Favorite Wife  1940  Cary Grant  actor   Nick Arden  2.0
871433    Penny Serenade  1941  Cary Grant  actor  Roger Adams  2.0

### List the leading roles that Cary Grant played in the 1940s in order by year.

In [121]:
c=cast[cast.name=='Cary Grant']
x=c[(c['year'] >= 1940) & (c['year'] <= 1949)]
x[x['n']>0].sort_values('year')

title  year        name   type  \
871448               The Howards of Virginia  1940  Cary Grant  actor   
871407                       His Girl Friday  1940  Cary Grant  actor   
871423                      My Favorite Wife  1940  Cary Grant  actor   
871450                The Philadelphia Story  1940  Cary Grant  actor   
871438                             Suspicion  1941  Cary Grant  actor   
871433                        Penny Serenade  1941  Cary Grant  actor   
871452                  The Talk of the Town  1942  Cary Grant  actor   
871429                 Once Upon a Honeymoon  1942  Cary Grant  actor   
871398                     Destination Tokyo  1943  Cary Grant  actor   
871422                             Mr. Lucky  1943  Cary Grant  actor   
871430                      Once Upon a Time  1944  Cary Grant  actor   
871390                  Arsenic and Old Lace  1944  Cary Grant  actor   
871425             None But the Lonely Heart  1944  Cary Grant  actor   
871424                         Night and Day  1946  Cary Grant  actor   
871427                             Notorious  1946  Cary Grant  actor   
871444      The Bachelor and the Bobby-Soxer  1947  Cary Grant  actor   
871445                     The Bishop's Wife  1947  Cary Grant  actor   
871421  Mr. Blandings Builds His Dream House  1948  Cary Grant  actor   
871402          Every Girl Should Be Married  1948  Cary Grant  actor   
871411                I Was a Male War Bride  1949  Cary Grant  actor   

                                   character    n  
871448                           Matt Howard  1.0  
871407                          Walter Burns  1.0  
871423                            Nick Arden  2.0  
871450                    C. K. Dexter Haven  1.0  
871438                      Johnnie Aysgarth  1.0  
871433                           Roger Adams  2.0  
871452                          Leopold Dilg  1.0  
871429                 Patrick 'Pat' O'Toole  1.0  
871398                         Capt. Cassidy  1.0  
871422  Joe Adams -posing as Joe Bascopolous  1.0  
871430                           Jerry Flynn  1.0  
871390                     Mortimer Brewster  1.0  
871425                            Ernie Mott  1.0  
871424                           Cole Porter  1.0  
871427                                Devlin  1.0  
871444                           Dick Nugent  1.0  
871445                                Dudley  1.0  
871421                         Jim Blandings  1.0  
871402                     Dr. Madison Brown  1.0  
871411                 Captain Henri Rochard  1.0

### How many roles were available for actors in the 1950s?

In [135]:
x=cast[(cast['year'] >= 1950) & (cast['year'] <= 1959)]
y=x[x.type=='actor']
y.character.nunique()

42734

**42734 roles were available**

### How many roles were available for actresses in the 1950s?

In [145]:
x=cast[(cast['year'] >= 1950) & (cast['year'] <= 1959)]
y=x[x.type=='actresses']
y.character.nunique()

0

**There were no roles for actresses**

### How many leading roles (n=1) were available from the beginning of film history through 1980?

In [149]:
x=cast[cast['n']==1]
y=x[x.year<=1980]
y.character.nunique()

15768

**15768 unique leading roles were available from beginning to 1980**

### How many non-leading roles were available through from the beginning of film history through 1980?

In [150]:
x=cast[cast['n']>1]
y=x[x.year<=1980]
y.character.nunique()

135337

**There were 135337 non-leading roles till 1980**

### How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?

In [152]:
x=cast[cast['n'].isnull()]
y=x[x.year<=1980]
y.character.nunique()

69175

**There were 69175 minor roles**